src/components/select/select.ts

```diff
@@ -343,19 +342,19 @@ export class Select extends BaseInput<string[]> implements AfterViewInit, OnDest
 
       // If the user passed a cssClass for the select, add it
       selectCssClass += selectOptions.cssClass ? ' ' + selectOptions.cssClass : '';
-      overlay.setCssClass(selectCssClass);
+      this._overlay.setCssClass(selectCssClass);
 
-      overlay.addButton({
+      this._overlay.addButton({
         text: this.okText,
         handler: (selectedValues) => this.value = selectedValues
       });
 
     }
 
-    overlay.present(selectOptions);
+    this._overlay.present(selectOptions);
 
     this._fireFocus();
-    overlay.onDidDismiss((value: any) => {
+    this._overlay.onDidDismiss((value: any) => {
```

Comment:
Requested changes have been made. Kindly re-review the PR.

Also, it's not going to work out declaring `_overlay` as a ViewController. When assigning `this._overlay = overlay`, `overlay` could either be an ActionSheet, Alert, or Popover. ActionSheet and Alert implement ViewController, but Popover implements OverlayProxy. So declaring `_overlay` as a ViewController creates type errors because in the case of a Popover, `overlay` would be missing a lot of properties of a ViewController.